### ISSUE TITLE
Search filmroll does not update collection

### ISSUE THREAD

**Issue by Mark-64** [darktable-org/darktable#9931]
There are some gui glitches in the collections lib.
To reproduce:

1) Import a few film rolls
2) with DT closed, rename on purpose the folder of one film roll
3) Open DT, the renamed film roll is shown in the collections lib with a strikethrough
4) Right-click on the film roll and select "search film roll..."
5) Select the renamed folder

**Expected behavior**: according to the code in `view_popup_menu_onSearchFilmroll()`
```

      /* reset filter so that view isn't empty */
      dt_view_filter_reset(darktable.view_manager, FALSE);

      /* update collection to view missing filmroll */
      _lib_folders_update_collection(new_path);

      DT_DEBUG_CONTROL_SIGNAL_RAISE(darktable.signals, DT_SIGNAL_FILMROLLS_CHANGED);

```
The filmroll name in the treeview should no longer be strikethrough. The collection should not be empty, the found film roll should be visualized. 

**Observed behavior**: The filmroll name is still strikethrough and stays so until a DT restart. The collection is empty. The lib reset button does not work until one of the filmrolls is selected with a double click

**Comment by domosbg** [comment]
In Darktable 3.6 it is working as expected. In newer versions like 3.7.0+684 it doesn't work anymore. In Darktable 3.6 the new_path variable was set. But I didn't find the Pull request where it was changed.

**Comment by Mark-64** [comment]
> In Darktable 3.6 it is working as expected

Well, almost.
Roll name  gets updated in the treeview and strikethrough is removed. Reset button works.
However the collection is empty like in 3.7.
For sure it is close to the expected behavior.
I will try to bisect

**Comment by Mark-64** [comment]
Ok, this issue is made of three distinct problems.

**The first**: strikethrough doesn't get removed when the film roll is found.
This the the result of git bisect:
```
4a7507a01ab86480cd81d8acb768e3abc36ed048 is the first bad commit
commit 4a7507a01ab86480cd81d8acb768e3abc36ed048
Author: phweyland <philippe.weyland@laposte.net>
Date:   Mon Jul 5 12:28:06 2021 -0300

    collect: avoid folder status checking at display time
    
    fixes #9042
    
    - add the memory table film_folder
    - connect collect queries on it
    - to be done: refresh the table on folders status change

 src/common/collection.c | 33 +++++++++++++++++++++++++++++++++
 src/common/collection.h |  3 +++
 src/common/darktable.c  |  2 ++
 src/common/database.c   |  3 +++
 src/libs/collect.c      | 35 ++++++++++++++++++-----------------
 5 files changed, 59 insertions(+), 17 deletions(-)
```
@phweyland can you have a look ?

**The second**: after the film roll is found the module reset button gets in a non functional state and it unlocks only after force-selecting a film roll in the tree view

```
4fee593fe62a4a24bc77804e51238db167699b71 is the first bad commit
commit 4fee593fe62a4a24bc77804e51238db167699b71
Author: Diederik ter Rahe <dterrahe@yahoo.com>
Date:   Sun Aug 1 17:50:00 2021 +0100

    remove view key handlers and some focus grabs

 src/bauhaus/bauhaus.c  |  6 +++---
 src/control/control.c  | 22 ----------------------
 src/dtgtk/culling.c    | 10 ----------
 src/dtgtk/thumbtable.c |  3 ---
 src/gui/gtk.c          | 12 ------------
 src/views/view.c       | 14 --------------
 src/views/view.h       |  2 --
 src/views/view_api.h   |  2 --
 8 files changed, 3 insertions(+), 68 deletions(-)
```
@dterrahe this is for you

**The third**: after the filmroll is found the collection is empty, instead of  being reset to "all" or to the found filmroll.
This issue is already in 3.6 and i didn't bisect

**Comment by phweyland** [comment]
I knew that renaming in parallel of dt running is not and probably won't be well treated.
However the use  case you describe must work. I'll look at it. Thanks.

**Comment by dterrahe** [comment]
The second issue:

The reset button works (it clears the criteria field behind the dropdown) but the central view does not get updated. Apparently that is dependent on it having focus; if I click on the central view, it immediately updates. After the Input NG changes that take input handling away from view-specific code, my hope was that giving focus to the central view all over the place wasn't needed anymore, so that it doesn't make things stop working correctly if this is forgotten somewhere or if something else temporarily has focus. Also, stealing focus is just not polite, even if we have gotten used to it.

So my _preferred_ solution here, would be to have lighttable update the central view correctly even if it doesn't have focus and clean up possible other such dependencies elsewhere. I don't immediately have a idea where to start looking, as this isn't my code. There may of course be gtk reasons why this wouldn't work. @AlicVB what do you think?

Alternatively, I could just put back all the `gtk_widget_grab_focus(dt_ui_center(darktable.gui->ui));` calls, but the original comments say this is to help accels, which is not longer the case. And we would probably give up on cleaning focus related issues forever...

**Comment by 7h0mas-R** [comment]
Hi,
not sure if my problem belongs here - but it sounds as if it is related - let me know, if I should make a separate issue:
Darktable version: 4.2.0
OS: MacOS 12.6 Monterey

Reproduce:
1) Database on NAS or external drive
2) Stop DT, disconnect NAS
3) Start DT, previews of missing folders are there, but cannot be edited. Folders/film rolls are strikethrough (OK)
4) Reconnect drive
5) Previews of missing folders are there, AND can be edited. Folders/film rolls are STILL strikethrough (NOK)

Background:
I have most of my database stored on a Synology NAS in my home LAN. The NAS goes into sleep mode when inactive and only wakes up, when there is activity. The Macbook loses the connection after some time, after the NAS sleeps but I basically just have to click on the NAS symbol in Finder to wake up the drive and access the data.

Whenever I start DT while the Mac is not connected, it will show the data on the NAS as unavailable, indicated by strikethrough formatting of the film roll. I see the preview images but cannot edit them in the darkroom module (as expected).
Then I switch to Finder and wake up/ connect to the NAS. Switch back to DT, film rolls and folders are still strike-through, but now, when I select a preview, I can edit the photo in the darkroom. So DT is aware of the files now and can obviously access them - but the strikethrough formatting remains, incorrectly indicating, that the files are missing.

I found now way to refresh the Collection view, except by closing and restarting DT. It looks, like this issue could have the same root cause as this thread. Would be nice, if there would at least be a manual possibility to refresh without restart.
